### PR TITLE
Rename Logger.debug to enabled, have logging be off by default, add Logg...

### PR DESCRIPTION
**Pull request by humphd**
...er tests.

### CODE REVIEW

File: src/core/logger.js [hunk 1]
```diff
@@ -45,28 +46,26 @@
...
+     * Whether the logger is enabled or not.
      *
      * @param {Boolean} value: State of the logger.
      */
-    logger.debug = function( value ) {
-      if ( value !== undefined ) {
-        __debug = value;
-      } else {
+    Logger.enabled = function( value ) {
+      if ( value === undefined ) {
```
Comment (secretrobotron) [inline src/core/logger.js]: Can we just return the value regardless? Doesn't seem like there needs to be a restriction.

File: src/core/logger.js [hunk 2]
```diff
@@ -45,28 +46,26 @@
...
         return __debug;
       }
+      __debug = value;
```
Comment (secretrobotron) [inline src/core/logger.js]: Should this be cleansed a bit with `!!`?

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://butyr.org:8000/86b9b6f07905fe0/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://butyr.org:8000/2ba0daf3a67b3b4/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/86b9b6f07905fe0/output.txt

Total script time: 0.79 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://butyr.org:8000/2ba0daf3a67b3b4/output.txt

Total script time: 0.66 mins

Server is running on http://pr313.butyr.org/templates/test.html

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:80/186063eea2b235e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 2

Live output at: http://botio.butyr.org:80/d238f026f5aa8d9/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:80/186063eea2b235e/output.txt

Total script time: 0.33 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:80/d238f026f5aa8d9/output.txt

Total script time: 0.47 mins

Server is running on http://pr313.butyr.org/templates/test.html

**Pull request merged**